crates/aiken-lang/src/tipo/pattern.rs

```diff
@@ -141,11 +141,11 @@ impl<'a, 'b> PatternTyper<'a, 'b> {
         pattern: UntypedPattern,
         tipo: Rc<Type>,
         ann_type: Option<Rc<Type>>,
-        is_assignment: bool,
```

Comment:
I don't remember this being is_let, if I recall it was genuinely whether or not this is an assignment pattern. Not exclusively let.